detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ForbiddenClassNameSpec.kt

```diff
@@ -10,33 +10,48 @@ private const val FORBIDDEN_NAME = "forbiddenName"
 class ForbiddenClassNameSpec {
 
     @Test
-    fun `should report classes with forbidden names`() {
+    fun `should report classes with exactly matching forbidden names`() {
         val code = """
-            class TestManager {} // violation
-            class TestProvider {} // violation
-            class TestHolder
+            class Manager {} // violation
+            class Provider {} // violation
+            class Holder
         """.trimIndent()
         assertThat(
             ForbiddenClassName(TestConfig(FORBIDDEN_NAME to listOf("Manager", "Provider")))
                 .lint(code)
         ).hasSize(2)
     }
 
+    @Test
+    fun `should treat patterns as case-sensitive`() {
+        val code = """
+            class Manager {} // allowed, since it doesn't match the rule's capitalization
+            class Provider {} // violation
+            class Holder
+        """.trimIndent()
+        assertThat(
+            ForbiddenClassName(TestConfig(FORBIDDEN_NAME to listOf("manager", "Provider")))
+                .lint(code)
+        ).hasSize(1)
```

Comment:
```suggestion
        ).singleElement()
```

Currently we are trying to remove the `FindingsAssert`. This change would make the migration easier. See https://github.com/detekt/detekt/pull/8430